DEVGUIDE.md

```diff
@@ -113,6 +113,17 @@ If your changes involve modifying the list of language keywords in any way, (e.g
 
 This only works on Windows/.NETStandard framework, so changing this from any other platform requires editing and syncing all of the XLF files manually.
 
+## Updating baselines in tests
+
+Some tests use "baseline" files.  There is sometimes a way to update these baselines en-masse in your local build,
+useful when some change affects many baselines.  For example, in the 'fsharpqa' tests the baselines
+are updated using scripts or utilities that allow the following environment variable to be set:
+
+```
+set TEST_UPDATE_BSL=1
+```
+
```

Comment:
Should a link to https://github.com/dotnet/fsharp/tree/master/tests/fsharp#workflow-when-adding-or-fixing-tests be added as I gather this environment variable only affects fsharpqa tests and running from command line, while the section I'm pointing at is handling the other fsharp tests (which are generally run from the IDE or a test runner).

For context, the reason I made the directory list explicit in the update.base.line.with.actuals.fsx script is that it requires more attention when updating the baselines, always fearing something goes wrong with "en masse" updating depending the scenario.

Maybe a word of caution about doing "en masse" update of all baselines to be used mostly with a message change but not when code / logic in the compiler is changed as well which requires more careful review.

Reply:
Baselines should be updated carefully. Having a tool to batch update them makes updating baselines easier, but it may introduce bugs that no one notices due to incorrect baseline updates.

@dsyme can we make this a bit more surgical? perhaps a script with a specific path to the test case to update.

Reply:
I'll add that caution to the note.  But the best place to be vigilant about this is in code review - mass update of baselines should correspond to minimal compiler changes where possible.

Reply:
Have added this: 

> Updating baselines en-masse should be done very carefully and subject to careful code review.   Where possible the
compiler change causing the en-masse update should be isolated and minimized so it is obvious at review time that no other code generation chagnes will be caused.

I'm just documenting what's there for now, not seeking to improve it :)

Reply:
@smoothdeveloper We should probably just combine https://github.com/dotnet/fsharp/blob/master/tests/fsharp/readme.md into DEVGUIDE.md - I'd actually forgotten that file was there.  I guess it's the right place for this note though, I'll move it

Reply:
@dsyme, possibly, although I like the idea of having readme specific to folders, they show up when you browse the repository from github, making it more user friendly to people perusing in the repository, and it is clear that the contents apply to that part of the tree only.

A very long DEVGUIDE.md with lots of details about specific parts of the repository may be less inviting and lead to overlook the top most information.

I rather have a DEVGUIDE.md being an entry point and geared toward general development, and linking to specific areas rather than an ever expanding file containing every specific development use case, this should scale better.

It is subjective which one is best anyways but I hope it clarifies why the folder specific files came.

In any case, thanks for adding the "be careful" notice, I think that was the most important!